From deda784bfaa92de7d57fdb5c3155fef03bc652e8 Mon Sep 17 00:00:00 2001
From: Zhu Donghong <zhudonghong@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:48 +0800
Subject: [PATCH] ipmi: add ipmi driver support
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56UPH

--------------------------------

Provide the KCS and BT of IPMI system interfaces that system software
uses for communication with BMC. Since hardware may occur default errors
due to lack of BMC hardware, these interface devices are disabled by
default.

To enable IPMI support, one may enable supported BT/KCS interfaces and
also select CONFIG_IPMI_HANDLER=y, CONFIG_IPMI_DEVICE_INTERFACE=y and
CONFIG_IPMI_SI=y.

Signed-off-by: Zhu Donghong <zhudonghong@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/boot/dts/chip3.dts | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/sw_64/boot/dts/chip3.dts b/arch/sw_64/boot/dts/chip3.dts
index df7d6082c47c..676740bf03b1 100644
--- a/arch/sw_64/boot/dts/chip3.dts
+++ b/arch/sw_64/boot/dts/chip3.dts
@@ -179,6 +179,30 @@ lpc: lpc@0x8037 {
 
 		};
 
+		ipmi-kcs@0x8037 {
+			#address-cells = <2>;
+			#size-cells = <2>;
+			device_type = "ipmi";
+			compatible = "ipmi-kcs";
+			reg = <0x8037 0x10000ca2 0x0 0x10>;
+			reg-size = <1>;
+			reg-spacing = <1>;
+			reg-shift = <0>;
+			status = "disabled";
+		};
+
+		ipmi-bt@0x8037 {
+			#address-cells = <2>;
+			#size-cells = <2>;
+			device_type = "ipmi";
+			compatible = "ipmi-bt";
+			reg = <0x8037 0x100000e4 0x0 0x10>;
+			reg-size = <1>;
+			reg-spacing = <1>;
+			reg-shift = <0>;
+			status = "disabled";
+		};
+
 		gpio: gpio@8036 {
 			#address-cells = <2>;
 			#size-cells = <2>;
-- 
2.34.1

